apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/backend/ast/Ast.kt

```diff
@@ -61,7 +64,7 @@ internal data class CodeGenerationAst(
       val typeRef: TypeRef,
       val schemaTypename: String?,
       val fragmentAccessors: List<FragmentAccessor>,
-      val isTypeCase: Boolean,
+      val isShape: Boolean,
```

Comment:
not really sure what this name means, could you please provide more context?

Reply:
@martijnwalraven pointed out that `TypeCase` was originally meant for the whole `when` statement as in programming languages like modula (https://www.cs.purdue.edu/homes/hosking/m3/reference/typecase.html) and not to the individual sets of type conditions. 

I'm trying to find something that conveys this idea of "partition of possible types" and `Shape` seemed to fit the requirements. `Shape` is also mentioned in the spec: https://spec.graphql.org/draft/#SameResponseShape(). And I think that'll apply here as well. A shape is the shape of a response object (vs interfaces that are sub trees of shapes)

I'll add some Kdoc in the next PR.